packages/core/src/shipping/consignment-discount.ts

```diff
@@ -0,0 +1,22 @@
+enum ConsignmentDiscountType {
+    Automatic = 'AUTOMATIC',
+    Coupon = 'COUPON',
+}
+
+interface ConsignmentDiscountBase {
+    id: number;
+    amount: number;
+    type: ConsignmentDiscountType;
+}
+
+interface ConsignmentAutomaticDiscount extends ConsignmentDiscountBase {
+    type: ConsignmentDiscountType.Automatic;
+}
+
+interface ConsignmentCouponDiscount extends ConsignmentDiscountBase {
+    type: ConsignmentDiscountType.Coupon;
+    couponId: number;
+    couponCode: string;
+}
+
+export type ConsignmentDiscount = ConsignmentAutomaticDiscount | ConsignmentCouponDiscount;
```

Comment:
Your solution is easy to understand. I'd go brutal like below. 
```
export type ConsignmentDiscount = 
    | { id: number; amount: number; type: 'AUTOMATIC' }
    | { id: number; amount: number; type: 'COUPON'; couponId: number; couponCode: string };
```

Reply:
I think what @bc-mackxu has is good since then we can use `ConsignmentCouponDiscount` and `ConsignmentAutomaticDiscount` can be used separately as well.

The only change I will suggest is in Base interface have `type` as a generic, so we don't need to specify it again in others.

Reply:
Having enum issue when mocking data in checkout-js. I changed to fixed string in SDK, like this `type: 'AUTOMATIC'`

Reply:
Can you try if this works
```
interface ConsignmentDiscountBase<T> {
    id: number;
    amount: number;
    type: T;
}

interface ConsignmentAutomaticDiscount extends ConsignmentDiscountBase<'AUTOMATIC'> {
}

interface ConsignmentCouponDiscount extends ConsignmentDiscountBase<'COUPON'> {
    couponId: number;
    couponCode: string;
}

```

Reply:
Yes, this is working.